### PR TITLE
Transaction queue handle races

### PR DESCRIPTION
**Pull request by hackaugusto**
Review/Merge after: #2112 

This removes transactions from the queue which have been invalidated because of a race

### CODE REVIEW

**Comment by rakanalh** [comment]
Rebased on latest master.

**Comment by LefterisJP** [comment]
Reviewing this

File: raiden/transfer/node.py [hunk 1]
```diff
@@ -961,14 +961,73 @@ def is_transaction_effect_satisfied(chain_state, transaction, state_change):
...
+    # Most transactions cannot be invalidated by others. These are:
+    #
+    # - close transactions
+    # - settle transactions
+    # - batch unlocks
+    #
+    # Deposits and withdraws are invalidated by the close, but these are not
+    # made atomic through the WAL.
+
+    is_our_failed_update_transfer = (
```
Comment (LefterisJP) [inline raiden/transfer/node.py]: Perhaps I misunderstand but shouldn't you also handle other cases in here? For example the one that is described as a Valid example in the comment above with the close and deposit?

File: requirements.txt [hunk 2]
```diff
@@ -3,6 +3,7 @@ git+https://github.com/LefterisJP/pystun@develop#egg=pystun
 
 cachetools==2.1.0
 coincurve==7.1.0
+dataclasses==0.6
```
Comment (LefterisJP) [inline requirements.txt]: Where is this used? I don't see any `@dataclass` anywhere

File: raiden/transfer/node.py (hunk 1 above)
Comment (rakanalh) [reply raiden/transfer/node.py]: Correct. The issue here is that deposit is not a state change but rather goes through the proxies directly. So there's no `ContractSendChannelSetTotalDeposit` or anything similar that we can invalidate in this case.

Comment (LefterisJP) [reply raiden/transfer/node.py]: Okay. Sucks ... I don't remember why it isn't a state change. @hackaugusto Was there a reason for it? Or is this something we can do at some point in the future?

**Comment by LefterisJP** [comment]
As per our offline discussion, this is good to go. Just rebase away the dependency addition/removal

File: raiden/transfer/node.py (hunk 1 above)
Comment (hackaugusto) [reply raiden/transfer/node.py]: > Was there a reason for it?

Saving time

**Pull request merged**